compiler/rustc_parse/src/parser/path.rs

```diff
@@ -624,9 +624,18 @@ impl<'a> Parser<'a> {
             GenericArg::Const(self.parse_const_arg()?)
         } else if self.check_type() {
             // Parse type argument.
+            let is_const_fn = self.look_ahead(1, |t| t.kind == token::OpenDelim(token::Paren));
+            let mut snapshot = self.diagnostic_snapshot();
```

Comment:
Could you try using this other places in the parser where we make a snapshot manually?

Reply:
I replaced all `self.clone()` with `self.create_snapshot_for_diagnostic()`.